SystemManager/Src/SystemManager.cpp

```diff
@@ -0,0 +1,43 @@
+//
+// Created by Gagan Deep Singh on 2023-07-08.
+// Edited by Navtaj Hundal on 2023-07-23.
+//
+
+#include "SystemManager.hpp"
+#include "sbus_receiver.hpp"
+#include "sbus_defines.h"
+#include "rcreceiver_datatypes.h"
+#include "drivers_config.hpp"
+#include "independent_watchdog.h"
+#include "tim.h"
+
+SystemManager::SystemManager():
+    rcController_(&huart2),
+    throttleMotorChannel_(&htim2, TIM_CHANNEL_1),
+    yawMotorChannel_(&htim2, TIM_CHANNEL_2),
+    rollMotorChannel_(&htim2, TIM_CHANNEL_3),
+    pitchMotorChannel_(&htim2, TIM_CHANNEL_4),
+    watchdog_(&hiwdg)
+{}
+SystemManager::~SystemManager() {}
+
+void SystemManager::flyManually() {
+    for(;;){
+            updateRCInputs();
+            executeInputs();
+    }
```

Comment:
I understand the appeal of this clean code approach, but I'm not really a fan. It tells you what you're doing, but not how. The two functions you abstracted this into are not complex at all and the logic could absolutely be contained in this function.

The reason I say this is because in `updateRcInputs` you update the inputs, but we then have no clue what is done with them once they have been updated, and in `executeInputs` we update the motor channels, but that doesn't tell us where the values we are updating came from. 

It requires  someone who is new to the code to understand it ALL, and as this grows it will become very confusing. All that is to say, I don't think we need to abstract these into functions, and we can abstract them later once we notice there is code that is being re-used in order to simplify it. If you disagree with me, I would be happy to discuss more.

Reply:
That sounds good, I can just remove the abstraction